src/h3/api/basic_int/__init__.py

```diff
@@ -432,49 +432,50 @@ def h3shape_to_cells(h3shape, res):
     """
 
     # todo: not sure if i want this dispatch logic here. maybe in the objects?
-    if isinstance(h3shape, H3Poly):
-        poly = h3shape
+    if isinstance(shape, LatLngPoly):
+        poly = shape
         mv = _cy.polygon_to_cells(poly.outer, res, holes=poly.holes)
-    elif isinstance(h3shape, H3MultiPoly):
-        mpoly = h3shape
+    elif isinstance(shape, LatLngMultiPoly):
+        mpoly = shape
         mv = _cy.polygons_to_cells(mpoly.polys, res)
-    elif isinstance(h3shape, H3Shape):
-        raise ValueError('Unrecognized H3Shape: ' + str(h3shape))
+    elif isinstance(shape, H3Shape):
+        raise ValueError('Unrecognized Shape: ' + str(shape))
     else:
-        raise ValueError('Unrecognized type: ' + str(type(h3shape)))
+        raise ValueError('Unrecognized type: ' + str(type(shape)))
 
     return _out_collection(mv)
 
 
 def cells_to_h3shape(cells, tight=True):
```

Comment:
Suggestion: Something I've been doing lately is `cells_to_h3shape(cells, *, tight=True)` to enforce kwargs for optional arguments - mostly to avoid `cells_to_h3shape(cells, False)` at the callsite, where the boolean argument is now opaque and confusing.

Reply:
I like that better.